### DISCUSSION TITLE
Flash Loan Smart Contract

### DISCUSSION THREAD

**Discussion by ghost** [mrgnlabs/marginfi-v2#216] [Q&A]
Are there any example contracts for taking out a flash loan and interacting with a DEX such as Jupiter?

**Comment by LevBeta** [comment]
https://github.com/mrgnlabs/mrgn-ts/blob/main/packages%2Fmarginfi-client-v2%2Fexamples%2Fuse-flash-loans-2.ts

**Reply by Gonk2** [reply]
What about just initiating a flash loan without calling an API from Jupiter? 

Just the taking out of Solana and then buying a specified token?

**Reply by LevBeta** [reply]
It's the same way, the Jup API in that example is completility swapable.

Create a withdraw ix
`const withdrawIx = await marginfiAccount.makeWithdrawIx(withdrawAmount, usdcBank.address);`

Create a deposit ix
`const depositIx = await marginfiAccount.makeRepayIx(usdtAmountToRepay, usdtBank.address, true);`

Add the IX's that you wish to use the funds for in between of them, and you the transaction

```
const flashLoanTx = await marginfiAccount.buildFlashLoanTx({
    ixs: [...withdrawIx.instructions, whateverYouWantIX's, ...depositIx.instructions],
    addressLookupTableAccounts,
  });
```

**Reply by Gonk2** [reply]
Thank you for your help
